File: Queue/C/Array Implementation/queue.c
// Array implementation of queue

#include<stdio.h>
#define MAX 10
int q[MAX];
int front  = -1,rear = -1;

void insert(int x){
    if(rear == MAX-1){
        printf("Queue Overflow\n");
        return;
    }
    if(front == -1)
        front = 0;
    rear = rear+1;
    q[rear] = x;
}

int delete(){
    int x;
    if(front == -1 || front == rear +1){
        printf("Queue underflow\n");
        return;
    }
    x = q[front];
    front  = front +1;
    return x;
}

int peek(){
    int x;
    if(front == -1 || front  == rear+1){
        printf("Queue Underflow\n");
        return;
    }
    x = q[front];
    return x;
}

void display(){
    if(front == -1 || front == rear +1){
        printf("Queue Underflow\n");
        return;
    }
    for(int i = front;i<=rear;i++){
        printf("%d ",q[i]);
    }
    printf("\n");
    return;
    
}

int main(void){
    insert(1);
    insert(2);
    display();
    return 0;
}

File: Queue/Java/queue.java
// Queue FIFO - First In First Out 
class Queue{
    class Node{
        int data;
        Node next;
        public Node(int data){
            this.data = data;
        }
    }
    Node front = null;
    Node rear = null;
    public void display(){
        Node p = front;
        if(front == null){
            System.out.println("Queue Underflow");
            return;
        }
        while(p!=null){
            System.out.println(p.data);
            p = p.next;
        }
        return;
    }
    public void insert(int data){
        Node tmp = new Node(data);
        if(front == null){
            front = tmp;
        }
        tmp.next = null;
        if(rear != null){
            rear.next= tmp;
        }
        rear = tmp;
        return;
    }
    
}

public class Main{
    public static void main(String args[]){
        Queue q = new Queue();
        q.insert(1);
        q.insert(2);
        q.insert(3);
        q.display();
    }
}
